.github/workflows/plugin-update-check.yml

```diff
@@ -0,0 +1,62 @@
+on:
+  repository_dispatch:
+    types: [plugin_update_event]
+
+jobs:
+  plugin-update-check:
+    runs-on: ubuntu-latest
+    env:
+      # PAYLOAD will contain a commit sha, plugin repo name, and will be of type 'plugin_update_event'
+      # e.g.,
+      #  {
+      #    "event_type": "plugin_update_event",
+      #    "client_payload": {
+      #        "sha":"abcdef",
+      #        "repo":"vault-plugin-database-snowflake"
+      #    }
+      #  }
+      COMMIT_SHA: "${{github.event.client_payload.sha}}"
+      REPO_NAME: "${{github.event.client_payload.repo}}"
+      BRANCH_NAME: "plugin-update-${{github.event.client_payload.repo}}-${{github.event.client_payload.sha}}"
+    steps:
+      - run: echo "would use $COMMIT_SHA of $REPO_NAME"
+        # checkout
+      - uses: actions/checkout@v3 # should be a sha, but eh
+        # activate go
+      - uses: actions/setup-go@v4
+      - name: update plugin
+        run: |
+          go get "github.com/hashicorp/$REPO_NAME@$COMMIT_SHA"
+          go mod tidy
+      - name: detect changes
+        id: changes
+        run: |
+          echo "count=$(git status --porcelain=v1 2>/dev/null | wc -l)" >> "$GITHUB_OUTPUT"
+      - name: commit/push
+        if: steps.changes.outputs.count > 0
+        run: |
+          git config user.name hc-github-team-secure-vault-ecosystem
+          git config user.email hc-github-team-secure-vault-ecosystem@users.noreply.github.com
+          git add .
+          git commit -m "Automated dependency upgrades"
+          git push -f origin ${{ github.ref_name }}:"$BRANCH_NAME"
```

Comment:
I don't see where $BRANCH_NAME gets created?

Reply:
Branch name is set in the env block up in line 20

Reply:
I saw the name being populated, but where is the branch itself being created?  Don't you need a `git checkout -b`?

Reply:
Oh, I see, this is a git push invocation I wasn't familiar with.